mlflow/store/artifact/r2_artifact_repo.py

```diff
@@ -32,3 +33,8 @@ def parse_s3_compliant_uri(self, uri):
 
     def _get_s3_client(self, addressing_style="virtual"):
         return super()._get_s3_client(addressing_style=addressing_style)
+
+    def setup_s3_endpoint(self):
+        """Setup Cloudflare R2 backend to be endpoint_url, otherwise all s3 requests will go to AWS S3 by default"""
+        endpoint_url = f"https://{urllib.parse.urlparse(self.artifact_uri).netloc.split('@')[-1]}"
+        MLFLOW_S3_ENDPOINT_URL.set(endpoint_url)
```

Comment:
We probably shouldn't set this environment variable (which could affect other workflows that use S3 in the same process). Instead, could we pass the proper endpoint URL when instantiating the boto3 S3 client?

Reply:
true, if they registered a r2 backed model then try to register a s3 backed model this will fail

Reply:
updated

Reply:
Thanks for the update!